docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,581 @@
-      <HitsWrapper show={query.length > 0 && focus} asGrid={hitsAsGrid}>
-        {indices.map(({ name, title, hitComp }) => (
-          <Index key={name} indexName={name}>
-            <header>
-              <h3>{title}</h3>
-              <Stats />
-            </header>
-            <Results>
-              <Hits hitComponent={hitComps[hitComp](() => setFocus(false))} />
-            </Results>
-          </Index>
-        ))}
-        <PoweredBy />
-      </HitsWrapper>
-    </InstantSearch>
-  )
-}
-```
+Algolia has an upper bound of 10KB for an index entry. If you get the error `AlgoliaSearchError: Record at the position XX objectID=xx-xx-xx-xx-xx is too big size=xxxx bytes` it means you exceeded that limit. Note how the excerpts are pruned to 5000 characters in the query. Make sure you prune long fields and don't index unnecessary data.
 
-At the top, you import `InstantSearch` from [`react-instantsearch-dom`](https://community.algolia.com/react-instantsearch) which is the root component that allows your whole search experience to connect to Algolia's service. As the name suggests, `Index` allows you to tap into an individual index and `Hits` provides you with the data returned for a user's search input. Finally [`connectStateResults`](https://community.algolia.com/react-instantsearch/connectors/connectStateResults.html) wraps around custom React components and provides them with high-level stats about the current search state such as the query, the number of results and how long it took to fetch them.
+## Adding the user interface
 
-You then import the styled components that make up the UI and the `Input` component into which the user enters the query.
+Now that there is data in the index, it is time to build the user interface for searching. It will display as a magnifying glass icon button that, when clicked, expands into a form field. Search results will appear in a popover below the input field as the user types.
 
-```jsx:title=src/components/search/index.js
-import { Root, SearchBox, HitsWrapper, PoweredBy } from "./styles"
-import Input from "./Input"
+The guide will use the following frameworks:
+
+- [React InstantSearch](https://community.algolia.com/react-instantsearch), a component library provided by Algolia for easily building search interfaces.
+- [Algolia Search](https://www.npmjs.com/package/algoliasearch) provides the API client for calling Algolia.
+- [Styled Components](https://styled-components.com) for embedding the CSS in the code, integrated using the [Gatsby styled component plugin](/packages/gatsby-plugin-styled-components/).
+- [Styled Icons](https://styled-icons.js.org/) provides the magnifying glass icon for the search bar.
+
+Styled Components can also be replaced by any other CSS solution you prefer.
+
+Install these frameworks by running the following command:
+
+```shell
+npm install react-instantsearch-dom algoliasearch styled-components gatsby-plugin-styled-components @styled-icons/fa-solid
```

Comment:
Does `gtasby-plugin-styled-components` need to be included in the `gatsby-config` file? That was in the earlier version of this doc.

Reply:
Good point. Strangely, it seems to work fine without it (I've been testing without it), but it really ought to be there. I'll add it.

This does mean the tutorial makes you change the `gatsby-config` twice, but I think that's preferable to having to mention the styled components in the "configuring algolia" section.